meson.build

```diff
@@ -377,6 +370,21 @@ executable('dosbox', ['src/main.cpp', 'src/dosbox.cpp', version_file],
                           + internal_deps,
            include_directories : incdir,
            install : true)
+# create a library so we can test things inside DOSBOX dep path
+libdosbox = static_library('dosbox', ['src/main.cpp', 'src/dosbox.cpp', version_file],
+                        include_directories : incdir,
+                        dependencies : [atomic_dep, threads_dep, sdl2_dep, libloguru_dep] + internal_deps)
```

Comment:
There's a fair bit of duplication here and in the above executable.

libdosbox appears to be subset of the deps needed for the executable.

It is possible to use or reference the `dosbox_dep` inside the `executable` statement somehow?

Also, if `stdcppfs_dep` was added to the libdosbox deps, hen we wouldn't need to call it out separately in the unit tests meson file.